Packs/Jira/Integrations/JiraV3/JiraV3.py

```diff
@@ -0,0 +1,4381 @@
+                                        ' the jira-get-id-by-attribute command to get the user\'s Account ID.',
+                         'original_estimate': 'The original estimate of the Jira issue.'
+                         }
+DEFAULT_FETCH_LIMIT = 50
+DEFAULT_FIRST_FETCH_INTERVAL = '3 days'
+DEFAULT_FETCH_INTERVAL = 1  # Unit is in minutes
+DEFAULT_PAGE = 0
+DEFAULT_PAGE_SIZE = 50
+# Errors
+OAUTH2_AUTH_NOT_APPLICABLE_ERROR = ('This command is not applicable to run with a Jira On Prem instance since On Prem instances'
+                                    ' do not use OAuth2 for authorization.')
+ID_OR_KEY_MISSING_ERROR = 'Please provide either the issue ID or key.'
+EPIC_ID_OR_KEY_MISSING_ERROR = 'Please provide either the epic ID or key.'
+CLOSE_INCIDENT_REASON = 'Issue was marked as "Resolved", or status was changed to "Done"'
+MIRROR_DIRECTION_DICT = {
+    'None': None,
+    'Incoming': 'In',
+    'Outgoing': 'Out',
+    'Incoming And Outgoing': 'Both'
+}
+# This will be appended to the attachment's name when mirroring an attachment from XSOAR to Jira
+ATTACHMENT_MIRRORED_FROM_XSOAR = '_mirrored_from_xsoar'
+COMMENT_MIRRORED_FROM_XSOAR = 'Mirrored from Cortex XSOAR'
+
+
+# Exception Classes
+class RevokedAccessTokenError(Exception):
+    """This class is used to raise exceptions when the access token stored in the integration context has been revoked
+    or not valid anymore, so we could try to retrieve another access token using the refresh token or show an error message
+    to the user.
+    """
+
+    def __init__(self):
+        super().__init__(('The access token has been revoked or no authorization was done for your instance, if so,'
+                          ' please refer to the documentation to authorize your instance'))
+    pass
+
+
+class JiraBaseClient(BaseClient, metaclass=ABCMeta):
+    """
+    This class is an abstract class. By using metaclass=ABCMeta, we tell python that this class behaves as an abstract
+    class, where we want to define the definition of methods without implementing them, and the child classes will need to
+    implement these methods.
+    """
+
+    # This will hold a mapping between the issue fields arguments that is supplied by the user, and the path
+    # of them in the issue fields object when creating a new issue, in dotted string format. For example, when
+    # creating a new issue, we send to the API an issue fields object that holds data about the new issue, and if we want to add
+    # the project_key, then we supply it as: {fields: {project: {key: THE_PROJECT_KEY}}}
+    ISSUE_FIELDS_CREATE_MAPPER = {
+        'summary': 'fields.summary',
+        'project_key': 'fields.project.key',
+        'project_id': 'fields.project.id',
+        'issue_type_name': 'fields.issuetype.name',
+        'issue_type_id': 'fields.issuetype.id',
+        # 'project_name': 'fields.project.name',  # Does not work for Jira Cloud
```

Comment:
is project_name field should be commented out?

Reply:
Yes, since the API does not support the project_name, only the project_id, and project_key, I had to manually retrieve the project_name if the user supplied it.